### PR TITLE
Add scopes pages and refactor Expressions page

### PR DESCRIPTION
**Pull request by chanikag**
This is to fix https://github.com/wso2/docs-mi/issues/946

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: en/docs/reference/synapse-properties/scopes.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
+# Scopes
+
+In WSO2 Micro Integrator scopes refer to the different layers where the properties can be set when mediating messages. 
+
+## default scope
+
+When the scope of a property is `default`, its value is available throughout the mediation flow. In addition to the user-defined properties, you can retrieve the following special properties from the `default` scope.
+[Generic Properties]({{base_path}}/reference/mediators/property-reference/generic-properties) are available in the default scope.
+
+| Property name  | Return Value                                                                                                                                                                                    |
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:10]:
```suggestion
| Property name  | Return value                                                                                                                                                                                    |
```

File: en/docs/reference/synapse-properties/expressions.md [hunk 2]
```diff
@@ -1,376 +1,109 @@
...
-
-Syntax:
-
--   `           base64Decode(string encodedValue)          `
--   `           base64Decode(string           encodedValue           , string charset)          `
-    `                     `
+The most commonly used expression languages in WSO2 MI are [Synapse Variables](#synapse-variables), [JSONPath (for JSON)](#jsonpath-expressions), and [XPath (for XML)](#xpath-expressions).
 
-#### get-property() function
+## Synapse Variables
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:7]:
```suggestion
## Synapse variables
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+| ReplyTo        | ReplyTo header value as a String, or empty string («») if a ReplyTo address is not defined.                                                                                         |
+| MessageID      | A unique identifier (UUID) for the message as a String, or empty string («») if a MessageID is not defined. This ID is guaranteed to be unique.                                     |
+| FAULT          | TRUE if the message has a fault, or empty string if the message does not have a fault.                                                                                              |
+| MESSAGE_FORMAT | Returns pox, get, soap11, or soap12 depending on the message. If a message type is unknown this returns soap12                                                                      |
+| OperationName  | Operation name corresponding to the message. A proxy service with a WSDL can have different operations. If the WSDL is not defined, WSO2 MI defines fixed operation call `mediate`. |
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:20]:
```suggestion
| OperationName  | Operation name corresponding to the message. A proxy service with a WSDL can have different operations. If the WSDL is not defined, WSO2 MI defines a fixed operation called `mediate`. |
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+Examples:
+``` xml
+<property name="PROPERTY_1" value="VALUE_1" scope="default" type="STRING"/>
+<property name="PROPERTY_2" expression="get-property('PROPERTY_1')"/>
+```
+
+!!! Note
+    To access a property with the `default` cope inside the `mediate()` method of a mediator, you can include the following configuration in a custom mediator created using the [Class mediator]({{base_path}}/reference/mediators/class-mediator):
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:29]:
```suggestion
    To access a property with the `default` scope inside the `mediate()` method of a mediator, you can include the following configuration in a custom mediator created using the [Class mediator]({{base_path}}/reference/mediators/class-mediator):
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
+# Scopes
+
+In WSO2 Micro Integrator scopes refer to the different layers where the properties can be set when mediating messages. 
+
+## default scope
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:5]:
```suggestion
## `default` scope
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    ``` java
+    public boolean mediate(org.apache.synapse.MessageContext mc) {  
+        // Available in both in-sequence and out-sequenc  
+        String propValue = (String) mc.getProperty("PropName");  
+        System.out.println("SCOPE_SYNAPSE : " + propValue);  
+        return true;  
+    }
+    ```
+
+## axis2 scope
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:40]:
```suggestion
## `axis2` scope
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+        org.apache.axis2.context.MessageContext axis2MsgContext;  
+        axis2MsgContext = ((Axis2MessageContext) mc).getAxis2MessageContext();   
+        // Available only in the sequence the property is defined.  
+        String propValue = (String) axis2MsgContext.getProperty("PropName");  
+        System.out.println("SCOPE_AXIS2 : " + propValue);  
+        return true;  
+    } 
+    ```
+
+## transport scope
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:73]:
```suggestion
## `transport` scope
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+`get-property('transport', String propertyName)`
+
+Examples:
+``` xml
+<property name="PROPERTY_1" value="VALUE_1" scope="transport" type="STRING"/>
+<property name="PROPERTY_2" expression="get-property('transport',PROPERTY_1')"/>
+```
+
+## registry scope
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:89]:
```suggestion
## `registry` scope
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+`get-property('registry', String registryPath)`
+
+
+Examples:
+``` xml
+<property name="regProperty" expression="get-property('registry', 'gov:/data/xml/collectionx')"/>
+<property name="regResourceProperty" expression="get-property('registry', 'gov:/data/xml/collectionx@abc')"/>
+```
+
+## system scope
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:107]:
```suggestion
## `system` scope
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+
+When the scope of a property is `system`,
+it refers to [Java System properties](https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html).
+You can retrieve properties in system scope using the following syntax.
+
+`get-property('system', String propertyName)`
+
+## environment scope
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:115]:
```suggestion
## `environment` scope
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+
+You can retrieve environment variables using the following syntax.
+
+`get-property('env', String propertyName)`
+
+## file scope
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:121]:
```suggestion
## `file` scope
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+resources to be sent through the fault sequence when faults exist. A
+given property with the `         operation        ` scope only exists
+in a single request and can be accessed by a single resource. 
+
+The syntax of the function takes the following format.
+
+`get-property('operation', String propertyName)`
+
+
+## axis2-client
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:142]:
```suggestion
## `axis2-client` scope
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+
+`get-property('env', String propertyName)`
+
+## `file` scope
+
+You can retrieve properties defined in the `file.properties` configuration file using the following syntax. Properties in the file are reloaded periodically according to the time interval defined by the `file.properties.sync.interval` system property in seconds. If this interval is not defined, changes to the properties will not be reloaded automatically during runtime.
+
+`get-property('file', String propertyName)`
+
+## operation scope
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:127]:
```suggestion
## `operation` scope
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+<property name="PROPERTY_1" value="VALUE_1" scope="default" type="STRING"/>
+<property name="PROPERTY_2" expression="get-property('PROPERTY_1')"/>
+```
+
+!!! Note
+    To access a property with the `default` cope inside the `mediate()` method of a mediator, you can include the following configuration in a custom mediator created using the [Class mediator]({{base_path}}/reference/mediators/class-mediator):
+
+    ``` java
+    public boolean mediate(org.apache.synapse.MessageContext mc) {  
+        // Available in both in-sequence and out-sequenc
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:33]:
```suggestion
        // Available in both in-sequence and out-sequence  
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+
+!!! Note
+
+    To access a property with the `axis2` scope inside the `mediate()` method of a mediator, you can include the following configuration in a custom mediator created using the [Class mediator]({{base_path}}/reference/mediators/class-mediator):
+
+    ``` java
+    public boolean mediate(org.apache.synapse.MessageContext mc) {  
+        org.apache.axis2.context.MessageContext axis2MsgContext;  
+        axis2MsgContext = ((Axis2MessageContext) mc).getAxis2MessageContext();   
+        // Available only in the sequence the property is defined.
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:66]:
```suggestion
        // Available only in the sequence in which the property is defined.  
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+``` xml
+<property name="regProperty" expression="get-property('registry', 'gov:/data/xml/collectionx')"/>
+<property name="regResourceProperty" expression="get-property('registry', 'gov:/data/xml/collectionx@abc')"/>
+```
+
+## `system` scope
+
+When the scope of a property is `system`,
+it refers to [Java System properties](https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html).
+You can retrieve properties in system scope using the following syntax.
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:111]:
```suggestion
You can retrieve properties in the `system` scope using the following syntax.
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 17]
```diff
@@ -0,0 +1,158 @@
...
+
+You can retrieve properties defined in the `file.properties` configuration file using the following syntax. Properties in the file are reloaded periodically according to the time interval defined by the `file.properties.sync.interval` system property in seconds. If this interval is not defined, changes to the properties will not be reloaded automatically during runtime.
+
+`get-property('file', String propertyName)`
+
+## `operation` scope
+
+You can retrieve a property in the operation context level from the
+`         operation        ` scope. The properties within
+iterated/cloned message with the `         operation        ` scope are
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:131]:
```suggestion
the iterated/cloned message with the `         operation        ` scope are
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 18]
```diff
@@ -0,0 +1,158 @@
...
+## `file` scope
+
+You can retrieve properties defined in the `file.properties` configuration file using the following syntax. Properties in the file are reloaded periodically according to the time interval defined by the `file.properties.sync.interval` system property in seconds. If this interval is not defined, changes to the properties will not be reloaded automatically during runtime.
+
+`get-property('file', String propertyName)`
+
+## `operation` scope
+
+You can retrieve a property in the operation context level from the
+`         operation        ` scope. The properties within
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:130]:
```suggestion
`operation` scope. The properties within
```

Shall we fix the formatting here? There are multiple occurrences. Also, let's not break the lines like this. Pls check all the other places as well.

File: en/docs/reference/synapse-properties/scopes.md [hunk 19]
```diff
@@ -0,0 +1,158 @@
...
+In WSO2 Micro Integrator scopes refer to the different layers where the properties can be set when mediating messages. 
+
+## `default` scope
+
+When the scope of a property is `default`, its value is available throughout the mediation flow. In addition to the user-defined properties, you can retrieve the following special properties from the `default` scope.
+[Generic Properties]({{base_path}}/reference/mediators/property-reference/generic-properties) are available in the default scope.
+
+| Property name  | Return value                                                                                                                                                                                    |
+|----------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| To             | Incoming URL as a String, or empty string («») if a To address is not defined.                                                                                                      |
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:12]:
```suggestion
| `To`             | Incoming URL as a String, or empty string («») if a To address is not defined.                                                                                                      |
```

Shall we format the property names in the table?

File: en/docs/reference/synapse-properties/scopes.md [hunk 20]
```diff
@@ -0,0 +1,158 @@
...
+| Action         | SOAP Addressing Action header value as a String, or empty string («») if an Action is not defined.                                                                                  |
+| FaultTo        | SOAP FaultTo header value as a String, or empty string («») if a FaultTo address is not defined.                                                                                    |
+| ReplyTo        | ReplyTo header value as a String, or empty string («») if a ReplyTo address is not defined.                                                                                         |
+| MessageID      | A unique identifier (UUID) for the message as a String, or empty string («») if a MessageID is not defined. This ID is guaranteed to be unique.                                     |
+| FAULT          | TRUE if the message has a fault, or empty string if the message does not have a fault.                                                                                              |
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:18]:
```suggestion
| `FAULT`          | `TRUE` if the message has a fault, or empty string if the message does not have a fault.                                                                                              |
```

Shall we also format the possible values in the table?

File: en/docs/reference/synapse-properties/expressions.md [hunk 21]
```diff
@@ -1,376 +1,109 @@
...
-The `get-property()` function allows any XPath expression used in a configuration to look up information from the current message context. Using the [Property mediator]({{base_path}}/reference/mediators/property-mediator), you can retrieve properties from the message context and header.
+There is a set of predefined synapse variables that you can use
+to extract data from the message. These synapse variables in different [scopes]({{base_path}}/reference/synapse-properties/scopes) are as follows:
 
-The syntax of the function takes the following format.
+### $ctx
 
--   `          get-property(String propertyName)         `
--   `          get-property(String scope, String propertyName)         `
+The prefix used to get properties set at the [default scope]({{base_path}}/reference/synapse-properties/scopes/#default-scope). For example, to get the value of property with name `ERROR_MESSAGE`, use the expression `$ctx:ERROR_MESSAGE`.
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:14]: Shall we reword this as one of the following?

- For example, to get the value of the property with the `ERROR_MESSAGE` name, use the `$ctx:ERROR_MESSAGE` expression.

or

- For example, to get the value of the `ERROR_MESSAGE` property name, use the `$ctx:ERROR_MESSAGE` expression.

File: en/docs/reference/synapse-properties/expressions.md [hunk 22]
```diff
@@ -1,376 +1,109 @@
...
 
-!!! Info
-    When the result of an XPath evaluation results in a single XML node, the
-    evaluator will return the text content of this node by default
-    (equivalent of doing `/root/body/node/text()`). If you want to retrieve
-    the node itself, you can configure the [Enrich mediator]({{base_path}}/reference/mediators/enrich-mediator) as shown
-    in the following example.
+    Deploy the following API. For instructions, see [Creating a REST API]({{base_path}}/develop/creating-artifacts/creating-an-api/).
+    
+    Note the property, `<property name="stockprop" expression="$ctx:ERROR_MESSAGE"/>` in the fault sequence configuration. It is used to log the error message that occurs due to a mediation fault.
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:25]:
```suggestion
    The `<property name="stockprop" expression="$ctx:ERROR_MESSAGE"/>` property in the fault sequence configuration is used to log the error message that occurs due to a mediation fault.
```

File: en/docs/reference/synapse-properties/expressions.md [hunk 23]
```diff
@@ -1,376 +1,109 @@
...
-The syntax of the function takes the following format.
+### $ctx
 
--   `          get-property(String propertyName)         `
--   `          get-property(String scope, String propertyName)         `
+The prefix used to get properties set at the [default scope]({{base_path}}/reference/synapse-properties/scopes/#default-scope). For example, to get the value of property with name `ERROR_MESSAGE`, use the expression `$ctx:ERROR_MESSAGE`.
 
-The function accepts scope as an optional parameter. It retrieves a
-message property at the given scope, which can be one of the following.
+Similarly, you can use `$ctx` prefix with [Generic Properties]({{base_path}}/reference/mediators/property-reference/generic-properties).
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:16]:
```suggestion
Similarly, you can use the `$ctx` prefix with [Generic Properties]({{base_path}}/reference/mediators/property-reference/generic-properties).
```

File: en/docs/reference/synapse-properties/expressions.md [hunk 24]
```diff
@@ -1,376 +1,109 @@
...
-In this example, the property definition, `<property
-name="stockerrorprop" expression="$ctx:ERROR\_MESSAGE"/\>` is equivalent
-to `<property name="stockerrorprop"
-expression="get-property('ERROR\_MESSAGE')"/\>`.
+### $trp
 
-Similarly, you can use `$ctx` prefix with [Generic Properties]({{base_path}}/reference/mediators/property-reference/generic-properties).
+The prefix used to get the transport headers. For example, to get the
+transport header named Content-Type of the current message, use the
+expression **$trp:Content-Type** . HTTP transport headers are not
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:60]:
```suggestion
`$trp:Content-Type` expression . HTTP transport headers are not
```

File: en/docs/reference/synapse-properties/expressions.md (hunk 21 above)
Comment (chanikag) [reply en/docs/reference/synapse-properties/expressions.md:14]: No it won't express the fact that we highlight here.

File: en/docs/reference/synapse-properties/expressions.md [hunk 25]
```diff
@@ -1,376 +1,109 @@
...
-    expression="get-property('transport','Content-Type')"/\>`. Similarly, you
-    can use `$trp` prefix with [HTTP Transport
-    Properties]({{base_path}}/reference/mediators/property-reference/http-transport-properties) .
+    expression="get-property('transport','Content-Type')"/>`.
 
-#### $body
+### $body
 
-The SOAP 1.1 or 1.2 body element. For example, the expression `$body//getQuote` refers to the first `getQuote` element in a SOAP body, regardless of whether the message is SOAP-11 or SOAP-12. We have discussed an example below.
+The prefix used to get the message payload. For example, the expression `$body//getQuote` refers to the first `getQuote` element in a message body.
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:100]:
```suggestion
The prefix used to get the message payload. For example, the `$body//getQuote` expression refers to the first `getQuote` element in a message body.
```

File: en/docs/reference/synapse-properties/expressions.md [hunk 26]
```diff
@@ -379,196 +112,91 @@ The SOAP 1.1 or 1.2 body element. For example, the expression `$body//getQuote`
...
-    
-    Note the following message in the log.
-    
-    ``` java
-    [2013-03-18 14:14:16,356] INFO - LogMediator To: http://localhost:9000/services/SimpleStockQuoteService, WSAction: urn:getQuote, SOAPAction: urn:getQuote, ReplyTo: http://www.w3.org/2005/08/addressing/anonymous, MessageID: urn:uuid:8a64c9cb-b82f-4d6f-a45d-bef37f8b664a, Direction: request, stockprop = http://localhost:9000/services/SimpleStockQuoteService
-    ```
+### $axis2
 
-#### $axis2
+The prefix used to get the properties at the [axis2 scope]({{base_path}}/reference/synapse-properties/scopes/#axis2-scope). For example, to get the value axis2 scope property with name `REST_URL_POSTFIX`, use the expression `$axis2:REST_URL_POSTFIX`.
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:129]:
```suggestion
The prefix used to get the properties at the [axis2 scope]({{base_path}}/reference/synapse-properties/scopes/#axis2-scope). For example, to get the value of the `axis2` scope property with the `REST_URL_POSTFIX` name, use the `$axis2:REST_URL_POSTFIX` expression.
```

File: en/docs/reference/synapse-properties/expressions.md [hunk 27]
```diff
@@ -670,24 +309,24 @@ Prefix used to get a SOAP 1.1 or 1.2 envelope level element. For example, to get
...
+[JSONPath](http://goessner.net/articles/JsonPath/) expressions are used for navigating and querying JSON data. JSONPath allows you to extract specific data from JSON documents.
 
-JSONPath expressions are used within mediators like Log, Filter, Property, and Script mediators to access and manipulate JSON data.
-`json-eval` function in expressions is used to specify JSONPath
+JSONPath expressions are used within mediators like Log, Filter, Property, and Switch mediators to extract JSON data.
+`json-eval` function in expressions is used to specify JSONPath.
 
-- JSONPath expressions are written in a specific syntax to navigate through the JSON structure.
-- The syntax typically starts with $ to represent the root of the JSON document.
+- The syntax usually starts with $ to represent the root of the JSON document.
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:321]:
```suggestion
- The syntax usually starts with `$` to represent the root of the JSON document.
```

File: en/docs/reference/synapse-properties/expressions.md [hunk 28]
```diff
@@ -743,37 +382,199 @@ The following table summarizes sample JSONPath expressions and their outputs:
...
 json-eval($trp:propertyName.student.name)
 ```
 
-Learn more about [JSONPath syntax](http://goessner.net/articles/JsonPath/).
+## XPath expressions
+
+[XPath (XML Path Language)](https://www.w3schools.com/xml/xpath_intro.asp) is a language used to navigate through elements and attributes in an XML document. It allows you to identify and extract specific portions of XML data, making it ideal for working with complex XML payloads, which are common in service integration scenarios.
+
+!!! Note
+    The WSO2 Micro Integrator supports standard XPath functions and variables through its underlying XPath engine. It supports XPath 1.0 by default where as the support for XPath 2.0 can be introduced by adding the following property in `<MI_HOME>/conf/deployment.toml`.
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:416]:
```suggestion
    The WSO2 Micro Integrator supports standard XPath functions and variables through its underlying XPath engine. It supports XPath 1.0 by default whereas the support for XPath 2.0 can be introduced by adding the following property in `<MI_HOME>/conf/deployment.toml`.
```

File: en/docs/reference/synapse-properties/expressions.md [hunk 29]
```diff
@@ -743,37 +382,199 @@ The following table summarizes sample JSONPath expressions and their outputs:
...
+        <item id="102">
+            <name>Laptop</name>
+            <price>1200</price>
+        </item>
+    </items>
+    <total>1500</total>
+</order>
+```
+
+The following table summarizes sample Xpath expressions and their outputs:
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:449]:
```suggestion
The following table summarizes sample XPath expressions and their outputs:
```

File: en/docs/reference/synapse-properties/expressions.md [hunk 30]
```diff
@@ -743,37 +382,199 @@ The following table summarizes sample JSONPath expressions and their outputs:
...
+    ```
+
+### XPath extension functions
+
+In addition to standard [XPath functions](https://www.w3schools.com/xml/xsl_functions.asp), the Micro Integrator supports the following custom functions for
+working with XPath expressions:
+
+#### base64Encode() function
+
+The `base64Encode` function returns the base64-encoded value of the
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:528]:
```suggestion
The `base64Encode()` function returns the base64-encoded value of the
```

File: en/docs/reference/synapse-properties/expressions.md [hunk 31]
```diff
@@ -743,37 +382,199 @@ The following table summarizes sample JSONPath expressions and their outputs:
...
+-   `           base64Encode(string value)          `
+-   `           base64Encode(string value, string charset)          `
+
+Example:
+
+`<property name="ENCODED_VALUE" expression="base64Encode('This is a test value')"/>`
+
+#### base64Decode() function
+
+The `base64Decode` function returns the original value of the specified
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:542]:
```suggestion
The `base64Decode()` function returns the original value of the specified
```

File: en/docs/reference/synapse-properties/expressions.md [hunk 32]
```diff
@@ -743,37 +382,199 @@ The following table summarizes sample JSONPath expressions and their outputs:
...
+-   `          get-property(String propertyName)         `
+-   `          get-property(String scope, String propertyName)         `
+
+The function accepts [scope]({{base_path}}/reference/synapse-properties/scopes) as an optional parameter. It retrieves a
+message property at the given scope. If you provide only the property name without the scope, the default `synapse` scope will be used.
+
+
+#### url-encode() function
+
+The url-encode function returns the URL-encoded value of the specified
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:569]:
```suggestion
The `url-encode()` function returns the URL-encoded value of the specified
```

File: en/docs/reference/synapse-properties/scopes.md [hunk 33]
```diff
@@ -0,0 +1,158 @@
...
+
+## `default` scope
+
+When the scope of a property is `default`, its value is available throughout the mediation flow. In addition to the user-defined properties, you can retrieve the following special properties from the `default` scope.
+[Generic Properties]({{base_path}}/reference/mediators/property-reference/generic-properties) are available in the default scope.
+
+| Property name  | Return value                                                                                                                                                                                    |
+|----------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| To             | Incoming URL as a String, or empty string («») if a To address is not defined.                                                                                                      |
+| From           | From address as a String, or empty string («») if a From address is not defined.                                                                                                    |
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/scopes.md:13]:
```suggestion
| From           | From address as a string, or empty string («») if a From address is not defined.                                                                                                    |
```

Please check other places as well.

File: en/docs/reference/synapse-properties/expressions.md [hunk 34]
```diff
@@ -743,37 +382,199 @@ The following table summarizes sample JSONPath expressions and their outputs:
...
+<td>XPath includes various built-in functions to work with nodes and data. Refer to <a href="https://www.w3schools.com/xml/xsl_functions.asp">more functions</a>.</td>
+<td><pre><code>expression="sum(//ns:item/ns:price)" xmlns:ns="http://ws.apache.org/ns/synapse"</code></pre></td>
+<td><pre><code>1500.0</code></pre></td>
+</tr>
+</tbody>
+</table>
+
+??? note "How to retrieve the XML node itself instead of text content"
+    When the result of an XPath evaluation results in a single XML node, the
+    evaluator will return the text content of this node (equivalent of doing `/root/body/node/text()`). If you want to retrieve
```
Comment (DinithiDiaz) [inline en/docs/reference/synapse-properties/expressions.md:495]:
```suggestion
    evaluator will return the text content of this node (similar to executing `/root/body/node/text()`). If you want to retrieve
```

File: en/docs/reference/synapse-properties/scopes.md (hunk 33 above)
Comment (DinithiDiaz) [reply en/docs/reference/synapse-properties/scopes.md]: Shall we update the other places in the table as well?

File: en/docs/reference/synapse-properties/scopes.md (hunk 19 above)
Comment (DinithiDiaz) [reply en/docs/reference/synapse-properties/scopes.md]: Shall we update the other places in the table as well?

**Review by DinithiDiaz — APPROVED** [review]
